frontend/src/container/RoutingPolicies/useRoutingPolicies.ts

```diff
@@ -0,0 +1,238 @@
+		setPolicyDetailsModalState((prev) => ({
+			...prev,
+			isOpen: false,
+			mode: null,
+		}));
+	};
+
+	const handleDeleteModalOpen = (routingPolicy: RoutingPolicy): void => {
+		setSelectedRoutingPolicy(routingPolicy);
+		setIsDeleteModalOpen(true);
+	};
+
+	const handleDeleteModalClose = (): void => {
+		setSelectedRoutingPolicy(null);
+		setIsDeleteModalOpen(false);
+	};
+
+	// Create Routing Policy
+	const {
+		mutate: createRoutingPolicy,
+		isLoading: isCreating,
+	} = useCreateRoutingPolicy();
+
+	// Update Routing Policy
+	const {
+		mutate: updateRoutingPolicy,
+		isLoading: isUpdating,
+	} = useUpdateRoutingPolicy();
+
+	// Policy Details Modal Action (Create or Update)
+	const handlePolicyDetailsModalAction = (
+		mode: PolicyDetailsModalMode,
+		routingPolicyData: {
+			name: string;
+			expression: string;
+			channels: string[];
+		},
+	): void => {
+		if (mode === 'create') {
+			createRoutingPolicy(
+				{
+					payload: mapRoutingPolicyToCreateApiPayload(
+						routingPolicyData.name,
+						routingPolicyData.expression,
+						routingPolicyData.channels,
+					),
+				},
+				{
+					onSuccess: () => {
+						toast.success('Routing policy created successfully');
+						queryClient.invalidateQueries(REACT_QUERY_KEY.GET_ROUTING_POLICIES);
+						handlePolicyDetailsModalClose();
+					},
+					onError: () => {
+						toast.error('Failed to create routing policy');
+					},
+				},
+			);
+		} else if (mode === 'edit' && selectedRoutingPolicy) {
+			updateRoutingPolicy(
+				{
+					id: selectedRoutingPolicy.id,
+					payload: mapRoutingPolicyToUpdateApiPayload(
+						routingPolicyData.name,
+						routingPolicyData.expression,
+						routingPolicyData.channels,
+					),
+				},
+				{
+					onSuccess: () => {
+						toast.success('Routing policy updated successfully');
+						queryClient.invalidateQueries(REACT_QUERY_KEY.GET_ROUTING_POLICIES);
+						handlePolicyDetailsModalClose();
+					},
+					onError: () => {
+						toast.error('Failed to update routing policy');
+					},
+				},
+			);
+		}
+	};
+
+	// Policy Details Modal Action Loading (Creating or Updating)
+	const isPolicyDetailsModalActionLoading = useMemo(() => {
+		if (policyDetailsModalState.mode === 'create') {
+			return isCreating;
+		}
+		if (policyDetailsModalState.mode === 'edit') {
+			return isUpdating;
+		}
+		return false;
+	}, [policyDetailsModalState.mode, isCreating, isUpdating]);
+
+	// Delete Routing Policy
+	const {
+		mutate: deleteRoutingPolicy,
+		isLoading: isDeletingRoutingPolicy,
+	} = useDeleteRoutingPolicy();
+
+	const handleDeleteRoutingPolicy = (): void => {
+		if (!selectedRoutingPolicy) {
+			return;
+		}
+		deleteRoutingPolicy(selectedRoutingPolicy.id, {
+			onSuccess: () => {
+				toast.success('Routing policy deleted successfully');
+				queryClient.invalidateQueries(REACT_QUERY_KEY.GET_ROUTING_POLICIES);
+				handleDeleteModalClose();
+			},
+			onError: () => {
+				toast.error('Failed to delete routing policy');
```

Comment:
Same here. Let's the error message from API response either and if possible it should be done using the existing component (let me know if there is a modal version available that doesn't block whole screen, we don't want the one that blocks the entire screen).

Reply:
Yeah don't think any extra component is required. We can just access the errormessage here and show it in the toast itself.

```
onError: (error) => {
  toast.error(`Error: ${error.message}`);
},
```